ui/src/main/webapp/app/gaffer/results-service.js

```diff
@@ -60,17 +60,38 @@ angular.module('app').factory('results', ['events', function(events) {
                 if(result.class === "uk.gov.gchq.gaffer.data.element.Entity") {
                     if(result.vertex !== undefined && result.vertex !== '') {
                         incomingResults.entities.push(result);
-                        results.entities.push(result);
+                        var toAdd = true;
```

Comment:
Being super picky here but toAdd should probably be named something more descriptive like elementExists.

Reply:
Removed this in my newest push.